Homework_2/report/main.tex

```diff
@@ -0,0 +1,173 @@
+$$
+Označimo $l(x) = \left(\text{min}(1,3+x) - \text{max}(0,2+x)\right)$. Končen integral je tako
+$$
+F_x = \int_{-3}^{-1} \int_{-1}^{1} x \, l(x) (1-|y|) f(x,y) \, dy \, dx.
+$$
+
+Sedaj implementiramo funkcijo \texttt{simpson2D(n)}, ki sprejme število delitev $n$ (mora biti sodo za Simpsonovo metodo) in vrne silo v $x$ smeri. Ker je širina integracijskih intervalov v $x$ in $y$ enaka $2$, uporabimo enako velikost koraka $h = 2/n$ za obe smeri.
+V funkciji ustvarimo vektor uteži $w = [1, 4, 2, \ldots, 2, 4, 1]$, dolžine $n+1$, da lahko do iskanih uteži pri posameznih vozlih dostopamo s pomočjo indeksa. Funkcija je nato dvojna zanka po $x$ in $y$ koordinatah, kjer v zanki po $y$ iteriramo do polovice $n/2$ in nato pomnožimo rezultat z 2, saj je integrand soda funkcija v $y$ smeri. 
+Za vsako točko $(x,y)$ izračunamo vrednost integranda, ga pomnožimo z utežjo in dodamo k skupni vsoti sile. Na koncu pomnožimo vsoto s faktorjem $h^2/9$, kar nam da končno vrednost sile v $x$ smeri.
+
+Tekom postopka implementacije funkcije \texttt{simpson2D(n)} smo implementirali funkcije \texttt{simpson6D(n)}, ki je naivna implementacija 6-kratnega integrala, \texttt{simpson3D(n)}, ki je implementacija 3-kratnega integrala in \texttt{monte\_carlo\_integration(n)}, ki izračuna 6-kratni integral z Monte Carlo metodo.
+Te funkcije smo implementirali, saj smo sprva mislili, da bo ne bo potrebna redukcija 6-kratnega integrala v 3-kratnega in 2-kratnega. Sedaj te funkcije lahko uporabimo za primerjavo rezultatov.
+
+\section{Rezultati}
+Da dosežemo željeno natančnost, povečujemo $n$ in preverimo da je absolutna vrednost kvocienta razlike med dvema zaporednima izračunoma in vrednosti zadnjega izračuna manjša od $10^{-10}$. Pri $n=200000$ dobimo silo, zaokroženo na 10 decimalk, $F\approx-0.2479229692$. Minus označuje, da je sila med kockama privlačna.
+Pri tem je relativna natančnost $6.76 \cdot 10^{-11}$, kar je manj kot $10^{-10}$, zato lahko rezultat sprejmemo. Čas izvajanja je bil približno $1500$ sekund.
+
+\begin{figure}[H]
+    \centering
+    \includegraphics[width=0.75\textwidth]{figures/force_convergence.png}
+    \caption{Konvergenca različnih metod.}
+    \label{fig:convergence}
+\end{figure}
+Za konec še primerjamo konvergenco in časovno zahtevnost rezultatov različnih metod na različnih intervalih za $n$. Na sliki \ref{fig:convergence} so prikazane vrednosti sile za različne $n$.
+Kljub temu, da je metoda \texttt{simpson6D} najpočasnejša, izgleda, da ima najboljšo konvergenco. \texttt{simpson3D} in \texttt{simpson2D} imata bistveno večja odstopanja pri $n=2,6,10,14,18$, kar je verjetno posledica tega, da zaradi preoblikovanja rešujemo drugačen integral kot na začetku. Ima pa \texttt{simpson2D} boljše rezultate kot \texttt{simpson3D}, kar je pričakovano,
+saj je integral po $z$ že bil izračunan analitično. \texttt{monte\_carlo\_integration} je najslabša metoda, saj je nedeterministična in je odvisna od naključne izbire točk.
```

Comment:
Ne vidim, zakaj bi bila Monte-Carlo najslabša metoda. To, da je nedeterministična še ne pomeni, da je slabša od ostalih.  

Tale graf po mojem ne pove pravih stvari. Primerjati je treba število funkcijskih izračunov (oziroma čas), ki je potreben za doseženo natančnost. Primerjate jabolka in hruške.  Namesto n bi jaz raje na graf narisal napako.